addon/-private/system/store.js

```diff
@@ -2669,7 +2665,13 @@ function _commit(adapter, store, operation, snapshot) {
   promise = _guard(promise, _bind(_objectIsAlive, internalModel));
 
   return promise.then((adapterPayload) => {
-    store._adapterRun(() => {
+    /*
+      Note to future spelunkers hoping to optimize.
+      We rely on this `run` creating a nested run loop
```

Comment:
To be clear this may not be what backburner itself considers a nested run-loop, as it has its own concept of this. Internal BB handles nested run-loops [via a stack of run-loops](https://github.com/ebryn/backburner.js/blob/master/lib/backburner/index.js#L52-L58)


For example:

```js
Ember.run(() => Ember.run(() => /* nested run loop */));
```

Both have the same queues, and an inner one may capture things actually intended for the outer one (today). Which will have nested renders etc.

---

```js
Ember.run(() => store._backburner.run(() => /* nested but entirely independently run-loop */));
```

Two run loops, but different queues so Ember.run.schedules within the bb.run will escape to the outer one, but bb.schedules wil remain captured.

---


TL;DR words are hard, maybe we should re-jigger the words in this comment.